docs/cheaha/slurm/slurm_tutorial.md

```diff
@@ -300,6 +351,214 @@ $ sacct -j 27101430
+    ### The result is saved to a file with the same name as the original file,
+    ### but with a '.wordcount' extension in the same directory as the original file.
+    wc -w "$FILE" > "$DIRNAME/${BASENAME}.wordcount"
+    echo "Processed $FILE"
+else
+    echo "File not found: $FILE"
+fi
+```
+
+The script begins by setting up the Slurm array job with required resources. The working directory is defined, and the script uses the `find` command to search for files matching the pattern random_file_*.txt within that working directory and its subdirectories, passing the resulting list of file paths to the sort command to ensure they are processed in order. The file corresponding to the current Slurm array task ID is selected, and its directory and base name are extracted. If the file exists, the script counts the words in the file using the `wc -w` command and saves the word count to a new file with a `.wordcount` extension in the same directory. If the file is not found, an error message is displayed.
+
+Before you run the script `dynamic_file_word_count.job`, determine the number of files in the directory and its subdirectories using the following command. The command counts the number of files matching the pattern random_file_*.txt in the path`$HOME/example_4/input_files` directory and its subdirectories, and stores the result in the variable `MAX_TASKS`.
+
+```bash
+$MAX_TASKS=$(find $HOME/example_4/input_files -type f -name "random_file_*.txt" | wc -l)
+```
+
+Next, submit the array job using the command below, which creates an array of tasks from 1 to the value of MAX_TASKS, where each task corresponds to processing a different file listed in the array.
+
+```bash
+$sbatch --array=1-"$MAX_TASKS" dynamic_file_word_count.job
+```
+
+In the output below, each file was processed independently by a Slurm job array task. The task IDs 1, 2, 3, 4, and 5 correspond to the five different files being processed. For instance, here the Slurm Job ID is `31934540`. Each output file contains the word count for a specific text file handled by its respective Slurm array task.
+
+The `find` command in the output shows the word counts for each of the random_file*.txt files, where each `.wordcount` file contains the word count for its corresponding input file.
+
+```bash
+### Listing all the output files generated by the Slurm from the `logs` directory
+$ cd $HOME/example_4/logs/
+$ ls dynamic_file_word_count*.out
+dynamic_file_word_count_31934540_1.out  dynamic_file_word_count_31934540_3.out
+dynamic_file_word_count_31934540_2.out  dynamic_file_word_count_31934540_4.out
+dynamic_file_word_count_31934540_5.out
+
+### This command finds all files with names matching "random_file*.wordcount" in the specified directory and subdirectories.
+### For each file found, concatenate and display its contents (using 'cat') to show the word count and their file path.
+$ find $HOME/example_4/input_files/ -type f -name "random_file*.wordcount" -exec cat {} \;
```

Comment:
This one is trickier to use backslashes with. See if you can find one or two reasonable points to insert a backslash (if that's allowed with find, find is a weird command).